core/src/main/java/org/infinispan/container/versioning/irac/IracEntryVersion.java

```diff
@@ -119,14 +140,32 @@ public boolean equals(Object o) {
          return false;
       }
 
-      IracEntryVersion version = (IracEntryVersion) o;
+      IracEntryVersion other = (IracEntryVersion) o;
 
-      return vectorClock.equals(version.vectorClock);
+      return Arrays.equals(vectorClock, other.vectorClock);
    }
 
    @Override
    public int hashCode() {
-      return vectorClock.hashCode();
+      return Arrays.hashCode(vectorClock);
+   }
+
+   private static MapEntry[] toMapEntryArray(TreeMap<ByteString, TopologyIracVersion> map) {
+      int length = map.size();
+      MapEntry[] entries = new MapEntry[length];
+      int index = 0;
+      for (Map.Entry<ByteString, TopologyIracVersion> e : map.entrySet()) {
+         entries[index++] = new MapEntry(e.getKey(), e.getValue());
+      }
+      return entries;
+   }
+
+   private static TreeMap<ByteString, TopologyIracVersion> toTreeMap(MapEntry[] entries) {
```

Comment:
How often is this called and how many MapEntry instances do you expect there to be? I ask because we could fake the SortedMap interface (it only invokes size, comparator and iterator) so that insertion is O(N) instead of O(NlogN).

Reply:
all the time :) 
it is used to `merge()` and `increment()` versions and helps me reduce the complexity of the methods.
I can drop it and manipulate the arrays directly. WDYT?